File: win-prog/ProgWinVB/Tree View and List View/SplitThreeFrames/SplitThreeFrames.vb
'-------------------------------------------------
' SplitThreeFrames.vb (c) 2002 by Charles Petzold
'-------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SplitThreeFrames
    Inherits Form

    Shared Sub Main()
        Application.Run(new SplitThreeFrames())
    End Sub

    Sub New()
        Text = "Split Three Frames"

        Dim pnl As New Panel()
        pnl.Parent = Me
        pnl.Dock = DockStyle.Fill

        Dim split1 As New Splitter()
        split1.Parent = Me
        split1.Dock = DockStyle.Left

        Dim pnl1 As New Panel()
        pnl1.Parent = Me
        pnl1.Dock = DockStyle.Left
        pnl1.BackColor = Color.Lime
        AddHandler pnl1.Resize, AddressOf PanelOnResize
        AddHandler pnl1.Paint, AddressOf PanelOnPaint

        Dim pnl2 As New Panel()
        pnl2.Parent = pnl
        pnl2.Dock = DockStyle.Fill
        pnl2.BackColor = Color.Blue
        AddHandler pnl2.Resize, AddressOf PanelOnResize
        AddHandler pnl2.Paint, AddressOf PanelOnPaint

        Dim split2 As New Splitter()
        split2.Parent = pnl
        split2.Dock = DockStyle.Top

        Dim pnl3 As New Panel()
        pnl3.Parent = pnl
        pnl3.Dock = DockStyle.Top
        pnl3.BackColor = Color.Tan
        AddHandler pnl3.Resize, AddressOf PanelOnResize
        AddHandler pnl3.Paint, AddressOf PanelOnPaint

        pnl1.Width = ClientSize.Width \ 3
        pnl3.Height = ClientSize.Height \ 3
    End Sub

    Private Sub PanelOnResize(ByVal obj As Object, ByVal ea As EventArgs)
        DirectCast(obj, Panel).Invalidate()
    End Sub

    Private Sub PanelOnPaint(ByVal obj As Object, _
                             ByVal pea As PaintEventArgs)
        Dim pnl As Panel = DirectCast(obj, Panel)
        Dim grfx As Graphics = pea.Graphics

        grfx.DrawEllipse(Pens.Black, 0, 0, _
                         pnl.Width - 1, pnl.Height - 1)
    End Sub
End Class

File: win-prog/ProgWinCS/Tree View and List View/SplitThreeFrames/SplitThreeFrames.cs
//-----------------------------------------------
// SplitThreeFrames.cs © 2001 by Charles Petzold
//-----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SplitThreeFrames: Form
{
     public static void Main()
     {
          Application.Run(new SplitThreeFrames());
     }
     public SplitThreeFrames()
     {
          Text = "Split Three Frames";

          Panel panel      = new Panel();
          panel.Parent     = this;
          panel.Dock       = DockStyle.Fill;

          Splitter split1  = new Splitter();
          split1.Parent    = this;
          split1.Dock      = DockStyle.Left;

          Panel panel1     = new Panel();
          panel1.Parent    = this;
          panel1.Dock      = DockStyle.Left;
          panel1.BackColor = Color.Lime;
          panel1.Resize   += new EventHandler(PanelOnResize);
          panel1.Paint    += new PaintEventHandler(PanelOnPaint);

          Panel panel2     = new Panel();
          panel2.Parent    = panel;
          panel2.Dock      = DockStyle.Fill;
          panel2.BackColor = Color.Blue;
          panel2.Resize   += new EventHandler(PanelOnResize);
          panel2.Paint    += new PaintEventHandler(PanelOnPaint);

          Splitter split2  = new Splitter();
          split2.Parent    = panel;
          split2.Dock      = DockStyle.Top;

          Panel panel3     = new Panel();
          panel3.Parent    = panel;
          panel3.Dock      = DockStyle.Top;
          panel3.BackColor = Color.Tan;
          panel3.Resize   += new EventHandler(PanelOnResize);
          panel3.Paint    += new PaintEventHandler(PanelOnPaint);

          panel1.Width  = ClientSize.Width  / 3;
          panel3.Height = ClientSize.Height / 3;
     }
     void PanelOnResize(object obj, EventArgs ea)
     {
          ((Panel) obj).Invalidate();
     }
     void PanelOnPaint(object obj, PaintEventArgs pea)
     {
          Panel    panel = (Panel) obj;
          Graphics grfx  = pea.Graphics;

          grfx.DrawEllipse(Pens.Black, 0, 0, 
                           panel.Width - 1, panel.Height - 1);
     }
}
